test/e2e/utils/kubernetes.go

```diff
@@ -168,11 +168,22 @@ func (k TestKubernetes) CleanNamespaceAndLogOnPanic(namespace string) {
 
 	if CleanupInfinispan == "TRUE" || panicVal == nil {
 		ctx := context.TODO()
-		ExpectMaybeNotFound(k.Kubernetes.Client.DeleteAllOf(ctx, &ispnv2.Batch{}, opts...))
+		if spacLabel == nil {
+			ExpectMaybeNotFound(k.Kubernetes.Client.DeleteAllOf(ctx, &ispnv2.Batch{}, opts...))
```

Comment:
You need to define Delete twice, you just need to define `opts` for different cases and keep delete function in the single place